### DISCUSSION TITLE
get only the info of a specific user

### DISCUSSION THREAD

**Discussion by ninjaibiss** [saul/demofile#270] [Questions] [answered]
Hi can someone help me with this I already tried the previous one but it seems to not work either I've been using a normal demo and not an overwatch demo

**Comment by saul** [comment]
Please can you explain in a bit more detail what you're trying to achieve?

**Comment by ninjaibiss** [comment]
a specific demo player stats like deaths kills and assists and score im using it for discord demo info

**Comment by ninjaibiss** [comment]
and how do we parse the demo using the share code without downloading it?

**Comment by ninjaibiss** [comment]
or is there any way on getting the download url using the demo sharecode

**Comment by saul** [comment, marked as answer]
It looks like https://github.com/akiver/CSGO-Demos-Manager can be used to download demos given a demo sharecode. From my understanding, it works by:

1. Decoding the share code - see https://github.com/akiver/CSGO-Demos-Manager/blob/7abb325ad3663732ca585addee52383a78751314/Core/ShareCode.cs#L74-L112
1. Passing the MatchId, OutcomeId and TokenId to `boiler-writter` - see https://github.com/akiver/CSGO-Demos-Manager/blob/9214dcc0c775c1460bbd4172af8c8289d181e55f/Services/Concrete/SteamService.cs#L218
1. `boiler-writter` requests full game info from Steam GC - see https://github.com/akiver/boiler-writter/blob/master/boiler-writter/CSGOMatchList.cpp#L38-L43
1. `boiler-writter` then writes the match list data (protobuf) to a file - see https://github.com/akiver/boiler-writter/blob/04cec2bf762dc33e3563e2fabfa8a4867ddaee4a/boiler-writter/main.cpp#L17-L24
1. CSGO-Demos-Manager then reads the match list data protobuf - see https://github.com/akiver/CSGO-Demos-Manager/blob/cb5cbe51789f6399ff73d0fe7f2afa368c3b27a2/Services/Concrete/DemosService.cs#L368-L397

It looks like the download URL can be found in `CMsgGCCStrike15_v2_MatchList.matches[].roundstatsall[].map`.

Unfortunately it would be way out of scope to do something similar in this library, so you're best off using CSGO-Demos-Manager for that.
